plugin/transfer/transfer.go

```diff
@@ -123,7 +126,11 @@ func (t *Transfer) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Ms
 		}
 		rrs = append(rrs, records...)
 		if len(rrs) > 500 {
-			ch <- &dns.Envelope{RR: rrs}
+			select {
+			case ch <- &dns.Envelope{RR: rrs}:
+			case <-closeCh:
+				return dns.RcodeServerFailure, ErrWriteMessage
```

Comment:
this would then returned the saved error from above

Reply:
https://github.com/coredns/coredns/blob/ea692844b17926ffae7bfdc9066c8abb89d5a1fc/plugin/transfer/transfer.go#L110-L117

change channel type from struct{} to error, and send  tr.Out result to channel.